### PR TITLE
Add threat intel mode property for applicable firewall resources

### LINKED ISSUE #638: Add Azure Firewall Threat Intel Mode settings
### Describe the feature end to end, including deployment scenario details under which the feature would occur.

The current module deploys the Azure Firewall with the threat intelligence mode set to alert only.

Request to add params and resource code to enable choice of other modes.

### Why is this feature important. Describe why this would be important for your organization and others. Would this impact similar orgs in the same way?

Security requirements to set to alert and deny mode.

### Please provide the correlation id associated with your error or bug.

xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx

### Can you describe any alternatives that you have taken since this feature does not exist?

_No response_

### Feature Implementation

```Shell
Update to vwanConnectivity.bicep

PARAMS SECTION
@sys.description('The Azure Firewall threat Intelligence mode to enable.')
@allowed([
  'Alert'
  'Deny'
  'Off'
])
param parThreatIntelMode string = 'Alert'

RESOURCE

resource resFirewallPolicies 'Microsoft.Network/firewallPolicies@2023-02-01' = if (parVirtualHubEnabled && parVirtualWanHubs[0].parAzFirewallEnabled) {
  name: parAzFirewallPoliciesName
  location: parLocation
  tags: parTags
  properties: (parAzFirewallTier == 'Basic') ? {
    sku: {
      tier: parAzFirewallTier
    }
  } : {
    dnsSettings: {
      enableProxy: parAzFirewallDnsProxyEnabled
      servers: parAzFirewallDnsServers
    }
    sku: {
      tier: parAzFirewallTier
    }
    threatIntelMode: parThreatIntelMode
  }
}
```

### Check previous GitHub issues

- [X] I have searched the issues for this item and found no duplicate

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by oZakari**
# Overview/Summary

Resolves #638 
[AB#31239](https://dev.azure.com/CSUSolEng/2a9d5a5a-8998-4ad0-81c8-ef3045e4da97/_workitems/edit/31239)
Adds Threat Intelligence Mode property to applicable Azure Firewall resources.

## This PR fixes/adds/changes/removes

1. Adds ability to set the threat intelligence mode to either Alert, Deny, or Off.

### Breaking Changes

1. No breaking changes as the default value is still the same as before this addition.

## Testing Evidence

Replace this with any testing evidence to show that your Pull Request works/fixes as described and planned (include screenshots, if appropriate).

## As part of this Pull Request I have

- [x] Read the [Contribution Guide](https://github.com/Azure/ALZ-Bicep/wiki/Contributing) and ensured this PR is compliant with the guide
- [x] Ensured the resource API versions in `.bicep` file/s I am adding/editing are using the latest API version possible
- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/ALZ-Bicep/pulls)
- [x] Associated it with relevant [GitHub Issues](https://github.com/Azure/ALZ-Bicep/issues)
- [ ] *(ALZ Bicep Core Team Only)* Associated it with relevant [ADO Items](https://aka.ms/alz/bicep/backlog)
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/ALZ-Bicep/tree/main)
- [ ] Performed testing and provided evidence.
- [ ] Updated one or more of the following tests *(if required)*
  - [Unit](https://github.com/Azure/ALZ-Bicep/blob/main/.github/workflows/bicep-build-to-validate.yml)
  - [Linting](https://github.com/Azure/ALZ-Bicep/tree/main/.github/workflows)
  - [E2E (End-To-End)](https://github.com/Azure/ALZ-Bicep/blob/main/tests/pipelines/bicep-build-to-validate.yml)
  - [ValidateAzCloud (Base validation in Azure Cloud)](https://github.com/Azure/ALZ-Bicep/blob/main/tests/pipelines/base-unit-validate.yml)
  - [ValidateMcCloud (Base validation in Azure China Cloud)](https://github.com/Azure/ALZ-Bicep/blob/main/tests/pipelines/mc-base-unit-validate.yml)
- [x] Updated relevant and associated documentation (e.g. Contribution Guide, Module READMEs, Wiki Docs etc.)
- [ ] If relevant, created or updated Code Tours [here](https://github.com/Azure/ALZ-Bicep/blob/main/.vscode/tours)

### CODE REVIEW

**Comment by oZakari** [comment]
@jtracey93 still need to test so leaving in draft for now.

File: infra-as-code/bicep/modules/hubNetworking/hubNetworking.bicep [hunk 1]
```diff
@@ -750,6 +759,7 @@ resource resAzureFirewall 'Microsoft.Network/azureFirewalls@2023-02-01' = if (pa
     firewallPolicy: {
       id: resFirewallPolicies.id
     }
+    threatIntelMode: parThreatIntelMode
```
Comment (jtracey93) [inline infra-as-code/bicep/modules/hubNetworking/hubNetworking.bicep:762]:
```suggestion
```

File: infra-as-code/bicep/modules/vwanConnectivity/vwanConnectivity.bicep [hunk 2]
```diff
@@ -327,6 +336,7 @@ resource resAzureFirewall 'Microsoft.Network/azureFirewalls@2023-02-01' = [for (
     firewallPolicy: {
       id: (parVirtualHubEnabled && hub.parAzFirewallEnabled) ? resFirewallPolicies.id : ''
     }
+    threatIntelMode: parThreatIntelMode
```
Comment (jtracey93) [inline infra-as-code/bicep/modules/vwanConnectivity/vwanConnectivity.bicep:339]:
```suggestion
```

**Review by jtracey93 — CHANGES_REQUESTED** [review]

**Pull request merged**